svc/shows/endpoint.go

```diff
@@ -166,3 +250,204 @@ func MakeGetShowsByCategoryEndpoint(s service, v validator.ValidateFunc) endpoin
 }
+
+// MakeAddShowEndpoint ...
+func MakeAddShowEndpoint(s service, v validator.ValidateFunc) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(AddShowRequest)
+		if err := v(req); err != nil {
+			return nil, err
+		}
+
+		err := s.AddShow(ctx, Show{
+			Title:         req.Title,
+			Cover:         req.Cover,
+			HasNewEpisode: req.HasNewEpisode,
+			Category:      req.Category,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		return true, nil
+	}
+}
+
+// MakeUpdateShowEndpoint ...
+func MakeUpdateShowEndpoint(s service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(UpdateShowRequest)
+
+		id, err := uuid.Parse(req.ID)
+		if err != nil {
+			return nil, fmt.Errorf("could not get show id: %w", err)
+		}
+
+		err = s.UpdateShow(ctx, Show{
+			ID:            id,
+			Title:         req.Title,
+			Cover:         req.Cover,
+			HasNewEpisode: req.HasNewEpisode,
+			Category:      req.Category,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		return true, nil
+	}
+}
+
+// MakeDeleteShowByIDEndpoint ...
+func MakeDeleteShowByIDEndpoint(s service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		id, err := uuid.Parse(request.(string))
+		if err != nil {
+			return nil, fmt.Errorf("could not get show id: %w", err)
+		}
+
+		err = s.DeleteShowByID(ctx, id)
+		if err != nil {
+			return nil, err
+		}
+
+		return true, nil
+	}
+}
+
+// MakeAddEpisodeEndpoint ...
+func MakeAddEpisodeEndpoint(s service, v validator.ValidateFunc) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(AddEpisodeRequest)
+		if err := v(req); err != nil {
+			return nil, err
+		}
+
+		showID, err := uuid.Parse(req.ShowID)
+		if err != nil {
+			return nil, fmt.Errorf("could not get show id: %w", err)
+		}
+
+		err = s.AddEpisode(ctx, Episode{
+			ShowID:        showID,
+			EpisodeNumber: req.EpisodeNumber,
+			Cover:         req.Cover,
+			Title:         req.Title,
+			Description:   req.Description,
+			ReleaseDate:   req.ReleaseDate,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		return true, nil
+	}
+}
+
+// MakeUpdateEpisodeEndpoint ...
+func MakeUpdateEpisodeEndpoint(s service, v validator.ValidateFunc) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(UpdateEpisodeRequest)
+		if err := v(req); err != nil {
+			return nil, err
+		}
+
+		id, err := uuid.Parse(req.ID)
+		if err != nil {
+			return nil, fmt.Errorf("could not get episode id: %w", err)
+		}
+
+		showID, err := uuid.Parse(req.ShowID)
+		if err != nil {
+			return nil, fmt.Errorf("could not get show id: %w", err)
+		}
+
+		err = s.UpdateEpisode(ctx, Episode{
+			ID:            id,
+			ShowID:        showID,
+			EpisodeNumber: req.EpisodeNumber,
+			Cover:         req.Cover,
```

Comment:
maybe if we don't validate Cover as required and non-empty we should check if it's not == "", then update it as well, else without it?